core/utility/include/options.h

```diff
@@ -20,19 +24,21 @@ namespace lightdb {
 
 
         std::optional<TValue> get(const TKey& key) const {
-            auto value = this->find(key);
-            return value != this->end()
+            auto value = _internalMap.find(key);
+            return value != _internalMap.end()
                 ? std::optional<TValue>{(*value).second}
                 : std::optional<TValue>{};
         }
 
         template<typename TOptional>
         std::optional<TOptional> get(const TKey& key) const {
-            auto value = this->find(key);
-            return value != this->end()
+            auto value = _internalMap.find(key);
+            return value != _internalMap.end()
                    ? std::optional<TOptional>{std::any_cast<TOptional>((*value).second)}
                    : std::optional<TOptional>{};
         }
+    private:
```

Comment:
This class shouldn't both derive from undered_map and contain an internal instance of one.  Remove the private field and delegate to the base constructor.